File: _LibSDK/LibSDKDemo/src/main/java/com/demo/fragment/FunctionFragment.java
package com.demo.fragment;

import android.os.Bundle;
import android.view.View;

import com.demo.activity.LeakCanaryActivity;
import com.demo.application.MainApplication;
import com.yline.application.SDKManager;
import com.yline.log.LogFileUtil;
import com.yline.test.BaseTestFragment;

public class FunctionFragment extends BaseTestFragment
{
	@Override
	public void testStart(View view, Bundle savedInstanceState)
	{
		// 测试 LeakCanaryActivity(不能放入LibSDK中,否则失效)
		addButton("LeakCanary Activity", new View.OnClickListener()
		{
			@Override
			public void onClick(View v)
			{
				LogFileUtil.v(MainApplication.TAG, "btn_leak_canary_activity");
				LeakCanaryActivity.actionStart(getContext());
			}
		});

		addButton("SDKManager", new View.OnClickListener()
		{
			@Override
			public void onClick(View v)
			{
				LogFileUtil.v(MainApplication.TAG, "btn_baseApplication");
				SDKManager.toast("测试，toast");
			}
		});

		addButton("CrashHandler", new View.OnClickListener()
		{
			@Override
			public void onClick(View v)
			{
				throw new ArithmeticException("crashHandler test");
			}
		});
	}
}

File: _LibSDK/LibSDKDemo/src/main/java/com/demo/fragment/FunctionFragment.kt
package com.demo.fragment

import android.os.Bundle
import android.os.Handler
import android.os.HandlerThread
import android.os.Process
import android.view.View
import com.demo.activity.LeakCanaryActivity
import com.yline.application.SDKManager
import com.yline.application.SDKManager.toast
import com.yline.log.LogUtil
import com.yline.test.BaseTestXFragment
import com.yline.utils.perf.MonitorUtil

class FunctionFragment : BaseTestXFragment() {

    override fun testStart(view: View, savedInstanceState: Bundle?) { // 测试 LeakCanaryActivity(不能放入LibSDK中,否则失效)
        addButton("LeakCanary Activity", View.OnClickListener {
            LogUtil.v("btn_leak_canary_activity")
            LeakCanaryActivity.actionStart(context)
        })

        addButton("SDKManager", View.OnClickListener {
            LogUtil.v("btn_baseApplication")
            toast("测试，toast")
        })

        addButton("CrashHandler", View.OnClickListener { throw ArithmeticException("crashHandler test") })
        addButton("MonitorUtil", View.OnClickListener { testMonitor() })
    }

    private fun testMonitor() {
        val result = MonitorUtil.getPidMemorySize(Process.myPid(), context)
        LogUtil.v("result = $result")
    }


}